File: Java/SpectableJavaExamplesTest/src/main/java/production/MatchCollection.java
package production;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

public class MatchCollection {
    public static final int MINIMUM = 0;
    public static final int MAXIMUM = 10000;

    private final List<Match> items = new ArrayList<>();

    public void add(Match item) {
        items.add(item);
    }

    public boolean delete(Match item) {
        return items.remove(item);
    }

    public List<Match> read() {
        return Collections.unmodifiableList(items);
    }

    public boolean update(Match oldItem, Match newItem) {
        int index = items.indexOf(oldItem);
        if (index < 0) return false;
        items.set(index, newItem);
        return true;
    }

    public int size() {
        return items.size();
    }
}

File: CSharp/SpecStudioExamplesTestsCSharp/production/MatchCollection.cs
namespace production
{
    using System.Collections.Generic;

    public class MatchCollection
    {
        public const int Minimum = 0;
        public const int Maximum = 10000;

        private readonly List<Match> _items = new();

        public void Add(Match item) => _items.Add(item);

        public bool Delete(Match item) => _items.Remove(item);

        public IReadOnlyList<Match> Read() => _items.AsReadOnly();

        public bool Update(Match oldItem, Match newItem)
        {
            int idx = _items.IndexOf(oldItem);
            if (idx < 0) return false;
            _items[idx] = newItem;
            return true;
        }

        public int Count => _items.Count;
    }
}
